bitcoin/src/blockdata/weight.rs

```diff
@@ -148,6 +163,9 @@ mod tests {
 
         let vb = Weight::from_vb(u64::MAX);
         assert_eq!(None, vb);
+
+        const WU: Weight = Weight::from_vb_const(1);
```

Comment:
@tcharding I know you hate the type annotation, however it seems to be required here.

Reply:
Yeah I wonder why this is needed everytime I have to write it.

Reply:
Can we have the test separate from the `from_vb` test, something like:
```rust

    #[test]
    fn from_vb_const() {
        const WU: Weight = Weight::from_vb_const(1);
        assert_eq!(Weight(4), WU);
    }
```

Reply:
> Yeah I wonder why this is needed everytime I have to write it.

Because the compiler can't infer the type :)

Reply:
> nit: Can we have the test separate from the from_vb test, something like:

done

Reply:
There's an open issue on rust-lang rfc 296 to allow type inference of const and statics.  Interestingly type inference is frowned upon in the google C++ style guide https://google.github.io/styleguide/cppguide.html#auto.  For concrete types it does seem possible, I think some people think it's more clear to have the type annotation, although it seem inconsistent to require it in some cases but not in others.

Reply:
Well, in C++ type inference is treacherous because C++ has very weakly typed numeric types, and has function/operator overloading (which makes type inference impossible to predict) and has inheritance (which is a less-bad version of the overloading problem).

But even in C++ there is no reason you should have have to write `CType ctype = new CType()`.

Reply:
Interesting, I didn't see mention of the reasoning for type annotation being required after searching around a bit.  Do you know where this is discussed out of curiosity?